cmd/internal/ec2/add.go

```diff
@@ -0,0 +1,174 @@
+package ec2
+
+import (
+	"embed"
+	"os"
+	"os/exec"
+	"os/user"
+	"strings"
+	"text/template"
+
+	"github.com/akitasoftware/akita-cli/printer"
+	"github.com/akitasoftware/akita-cli/telemetry"
+	"github.com/pkg/errors"
+)
+
+const (
+	envFileName         = "postman-lc-agent"
+	envFileTemplateName = "postman-lc-agent.tmpl"
+	envFileBasePath     = "/etc/default/"
+	envFilePath         = envFileBasePath + envFileName
+
+	serviceFileName     = "postman-lc-agent.service"
+	serviceFileBasePath = "/usr/lib/systemd/system/"
+	serviceFilePath     = serviceFileBasePath + serviceFileName
+)
+
+// Embed files inside the binary. Requires Go >=1.16
+
+//go:embed postman-lc-agent.service
+var serviceFile string
+
+// FS is used for easier template parsing
+
+//go:embed postman-lc-agent.tmpl
+var envFileFS embed.FS
+
+// Helper function for reporting telemetry
+func reportStep(stepName string) {
+	telemetry.WorkflowStep("Starting systemd conguration", stepName)
+}
+
+func setupAgentForServer(collectionId string) error {
+
+	err := checkUserPermissions()
+	if err != nil {
+		return err
+	}
+	err = checkSystemdExists()
+	if err != nil {
+		return err
+	}
+
+	err = configureSystemdFiles(collectionId)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func checkUserPermissions() error {
+	// TODO: Make this work without root
+
+	// Exact permissions required are
+	// read/write permissions on /etc/default/postman-lc-agent
+	// read/write permission on /usr/lib/system/systemd
+	// enable, daemon-reload, start, stop permission for systemctl
+
+	printer.Infof("Checking user permissions \n")
+	cu, err := user.Current()
+	if err != nil {
+		return errors.Errorf("could not get current user. OS returned error: %q \n", err)
+	}
+	if strings.EqualFold(cu.Name, "root") {
+		return errors.Errorf("Please run the command again with user: root. \n")
+	}
```

Comment:
I think an explanation of what we need to do with root permissions would be appropriate here, like the outline you introduced above.